src/checkers/inference/dataflow/InferenceTransfer.java

```diff
@@ -205,6 +209,11 @@ private InferenceAnalysis getInferenceAnalysis() {
             AnnotatedTypeMirror atm) {
 
         Slot slotToRefine = getInferenceAnalysis().getSlotManager().getVariableSlot(atm);
+        // Getting the declared type of a RefinementVariableSlot
+        // getRefined() should always return the slot of the declared type value
+        if (slotToRefine instanceof RefinementVariableSlot) {
```

Comment:
Also test ComparisonVariableSlot? like
https://github.com/txiang61/checker-framework-inference/blob/9c961c5541e217521e326514e2b6a5fbaac39c65/src/checkers/inference/dataflow/InferenceTransfer.java#L211

Reply:
Alternatively, we focus on directly getting left-hand-side ATM (i.e. declared annotated type)
we can use assertion like `assert !(slotToRefine instanceof RefinementVariableSlot)`. This would easily expose problems when we get the wrong atm.

Otherwise, we have to exhaustively explore the possibility of `slotToRefine ` being a RefinementVariableSlot or ComparisonVariableSlot.

Reply:
Not work! `getAnnotatedTypeLhs` can't work correctly in CFI. Because https://github.com/opprop/checker-framework-inference/blob/7c2611f14dd571ed17ac624c1ce94f299ef2a2c0/src/checkers/inference/dataflow/InferenceTransfer.java#L233